clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -335,6 +335,14 @@ public void poll(long timeout) throws IOException {
         // we use the time at the end of select to ensure that we don't close any connections that
         // have just been processed in pollSelectionKeys
         maybeCloseOldestConnection(endSelect);
+
+        Iterator<String> connectedIt = connected.iterator();
+        while (connectedIt.hasNext()) {
+            if (disconnected.containsKey(connectedIt.next())) {
+                log.warn("Channel {} is marked connected and disconnected at the same time");
+                connectedIt.remove();
```

Comment:
@ijuma I want to add a test, but unfortunately it is hard to mock such a network environment.
I have tested it many times in my company's  LVS Proxy env.
Do you have some suggestions about it?